src/components/item/sharing/shortLink/AliasValidation.tsx

```diff
@@ -0,0 +1,80 @@
+import { useEffect, useState } from 'react';
+
+import { FormHelperText, styled } from '@mui/material';
+
+import { ShortLinkAvailable } from '@graasp/sdk';
+
+import { useBuilderTranslation } from '@/config/i18n';
+import { BUILDER } from '@/langs/constants';
+import { isValidAlias } from '@/utils/shortLink';
+
+const WrapHelper = styled(FormHelperText)(() => ({
+  overflowWrap: 'anywhere',
+  whiteSpace: 'normal',
+  maxWidth: '240px',
+}));
+
+type Props = {
+  alias: string;
+  aliasUnchanged: boolean;
+  isAvailableLoading: boolean;
+  shortLinkAvailable?: ShortLinkAvailable;
+  onValidAlias: (isValid: boolean) => void;
+  onError: (hasError: boolean) => void;
+};
+
+const AliasValidation = ({
+  alias,
+  aliasUnchanged,
+  isAvailableLoading,
+  shortLinkAvailable,
+  onValidAlias,
+  onError,
+}: Props): JSX.Element => {
+  const { t: translateBuilder } = useBuilderTranslation();
+  const ALIAS_VALID_MSG = translateBuilder(BUILDER.ALIAS_VALID_MSG);
+  const ALIAS_UNCHANGED_MSG = translateBuilder(BUILDER.ALIAS_UNCHANGED_MSG);
+  const ALIAS_ALREADY_EXIST = translateBuilder(BUILDER.ALIAS_ALREADY_EXIST);
+
+  const [message, setMessage] = useState<string>(ALIAS_VALID_MSG);
+  const [valid, setValid] = useState<boolean>(true);
+  const [available, setAvailable] = useState<boolean>(false);
+
+  const hasError = !valid || (!isAvailableLoading && !available);
+
+  // check for the validity and availability of the alias
+  useEffect(() => {
+    const { isValid, messageKey, data } = isValidAlias(alias);
+    setValid(isValid);
+
+    if (!isValid) {
+      const msgKey = messageKey ?? BUILDER.SHORT_LINK_UNKOWN_ERROR;
+      setMessage(translateBuilder(msgKey, { data }));
+    } else if (aliasUnchanged) {
+      setMessage(ALIAS_UNCHANGED_MSG);
+      setAvailable(true);
+    } else if (isAvailableLoading) {
+      setMessage(translateBuilder(BUILDER.ALIAS_CHECKING_MSG));
+    } else {
+      const aliasAvailable = shortLinkAvailable?.available === true;
```

Comment:
I think I can't omit the true equality here, because I want aliasAvailable to be a boolean.
If I remove the `=== true` and the shortLinkAvailable is `null` or `undefined`, the aliasAvailable can have a boolean or undefined has value, but in my case I need a boolean.
```
type Available = { available: boolean };

const nullAvailable = null;
const available = { available: true };
const notAvailable = { available: false };

const logAvailableOrNull = (a: Available | null) => console.log(a?.available);

const logAvailableBooleanCheck = (a: Available | null) => console.log(a?.available === true);

console.log('--- logAvailableOrNull ---');
logAvailableOrNull(nullAvailable);
logAvailableOrNull(available);
logAvailableOrNull(notAvailable);

console.log('--- logAvailableBooleanCheck ---');
logAvailableBooleanCheck(nullAvailable);
logAvailableBooleanCheck(available);
logAvailableBooleanCheck(notAvailable);
```

Reply:
Then what about `Boolean(available)`?

Reply:
Nice, it's works !